### PR TITLE
Add postgres integration

### PR DESCRIPTION
**Pull request by saleel**
## Description

Add support for Posgres DB as persistance layer for API

## Related Issue

#85 

- Add Postgres support
- Set Postgres as default `DB_TYPE` (new env)
- Add Postgres seed script
- Update docker-compose to include Postgres
- Update Docker file for FE projects
- Fix auth issue in off-chain group flow in dashboard

## Does this introduce a breaking change?

-   [x] Yes
-   [ ] No

A new env is added to specify the DB storage layer `DB_TYPE`
By default this is set to Postgres and you are expected to also set `DB_URL` env with Postgres connection string. The DB also needed to be seeded with `./database/seed.sql` file

To continue using Sqlite (especially for easier development), set `DB_TYPE=sqlite` and `DB_URL=/path/to/sqlite_file.db`

### CODE REVIEW

File: apps/client/Dockerfile [hunk 1]
```diff
@@ -13,6 +13,8 @@ RUN yarn build:all
 
 # Create image for the app by copying build artifacts from builder
 FROM nginx:stable-alpine as runner
+RUN rm /etc/nginx/conf.d/default.conf
```
Comment (saleel) [inline apps/client/Dockerfile:16]: A fix in Dockerfile for FE projects
Not purely related to this PR, but found during testing docker flow

File: apps/dashboard/src/pages/my-groups.tsx [hunk 2]
```diff
@@ -41,26 +40,23 @@ export default function MyGroups(): JSX.Element {
...
-                    console.log("no jwt")
-                    setJwtInCookies(false)
+                    // Failure - Redirect to login page
+                    navigate("/sso")
                 })
         })()
-    }, [])
-
-    useEffect(() => {
-        if (jwtInCookies) {
```
Comment (saleel) [inline apps/dashboard/src/pages/my-groups.tsx:56]: Not sure how it was working - when the page loads it send a request to `getUser` and `jwtInCookies  = true`. But before this happen the other `useEffect` hook check for `jwtInCookies`, which would be false and redirect the user back to `/sso`
Fixed this my merging both hooks in to one.

File: docker-compose.yml [hunk 3]
```diff
@@ -1,24 +1,54 @@
...
-        env_file:
-            - .env
         environment:
             - NODE_ENV=production
-            - DB_NAME="/home/node/api/zk-groups.db"
-        volumes:
-            - sqlite:/home/node/api/zk-groups.db
+            - DB_TYPE=postgres
+            - DB_URL=postgres://root:helloworld@postgres:5432/zk-groups
+            - JWT_SECRET_KEY=zk_groups_jwt_secret
```
Comment (cedoor) [inline docker-compose.yml:33]: Why de we need to copy all the env variables here? Shouldn't be automatically loaded from the `.env` file?

File: docker-compose.yml [hunk 4]
```diff
@@ -1,24 +1,54 @@
 version: "3.9"
 
 services:
+    postgres:
+        container_name: postgres
+        image: postgres
+        environment:
+            POSTGRES_USER: root
```
Comment (cedoor) [inline docker-compose.yml:8]: Shouldn't user & password be loaded from the `.env` file?

File: docker-compose.yml (hunk 3 above)
Comment (saleel) [reply docker-compose.yml:33]: I specified it explicitly here for more visibility - mainly for Devops team. It not safe to load the secrets from .env file nor from docker-compose. We wont have .env file when the repo is cloned in EC2.

We can check with Ops team what would be the best way to handle secret envs. In Kubernetes there are ways to do this, not sure with our proposed setup. For now it was added here so docker-compose will have all the info needed for deployment. Can revert it if this is not helpful.

Comment (cedoor) [reply docker-compose.yml:33]: Ahhn I see. It makes sense then!

Comment (cedoor) [reply docker-compose.yml:33]: We should just remember to sync the env variables every time we change something

File: docker-compose.yml (hunk 4 above)
Comment (cedoor) [reply docker-compose.yml:8]: Yep, got it. Do you think we should have a specific `.env` variable for each app and a root `.env` for vars shared among many apps variables? It seems like NX recommend it: https://nx.dev/recipes/environment-variables/define-environment-variables

Comment (cedoor) [reply docker-compose.yml:8]: Yes, let me create it!

**Pull request merged**